dianna/methods/lime.py

```diff
@@ -149,3 +155,51 @@ def explain_image(self,
         mask = explanation.get_image_and_mask(label, positive_only=positive_only, hide_rest=hide_rest,
                                               num_features=num_features, **get_image_and_mask_kwargs)[1]
         return mask
+
+    def _prepare_image_data(self, input_data):
+        """
+        Transforms the data to be of the shape and type LIME expects
+
+        Args:
+            input_data (NumPy-compatible array): Data to be explained
+
+        Returns:
+            transformed input data, preprocessing function to use with utils.get_function()
+        """
+        input_data = utils.to_xarray(input_data, self.axes_labels, LIME.required_labels)
+        # remove batch axis from input data; this is only here for a consistent API
+        # but LIME wants data without batch axis
+        if not len(input_data['batch']) == 1:
+            raise ValueError("Length of batch axis must be one.")
+        input_data = input_data.sel(batch=0)
+        # ensure channels axis is last and keep track of where it was so we can move it back
+        channels_axis_index = input_data.dims.index('channels')
+        input_data = utils.move_axis(input_data, 'channels', -1)
+        # create preprocessing function that puts model input generated by LIME into the right shape and dtype,
+        # followed by running the user's preprocessing function
+        full_preprocess_function = self._get_full_preprocess_function(channels_axis_index, input_data.dtype)
+        # LIME requires float64 numpy data
+        return input_data.values.astype(np.float64), full_preprocess_function
+
+    def _get_full_preprocess_function(self, channel_axis_index, dtype):
+        """
+        Create a preprocessing function that incorporates both the (optional) user's
+        preprocessing function, as well as any needed dtype and shape conversions
+
+        Args:
+            channel_axis_index (int): Axis index of the channels in the input data
+            dtype (type): Data type of the input data (e.g. np.float32)
+
+        Returns:
+            Function that first ensures the data has the same shape and type as the input data,
+            then runs the users' preprocessing function
+        """
+        # LIME generates numpy arrays, so it is easiest here to use np.transpose to put the channels
+        # axis in the right place
+        # one is added to the channels axis index because there is an extra first axis: the batch axis
+        def moveaxis_function(data):
+            return np.moveaxis(data, -1, channel_axis_index+1).astype(dtype)
```

Comment:
```suggestion
            return np.moveaxis(data, -1, channel_axis_index + 1).astype(dtype)
```

Again the missing whitespace.